### PR TITLE
feat: new type rule for `try-with`

### PR DESCRIPTION
**Pull request by magnus-madsen**
The new type rule permits multiple effects:

```scala
eff Gen {
    pub def gen(x: Int32): Unit
}

eff Print {
    pub def print(x: String): Unit
}

def generator(x: Int32): Unit \ Gen + Print =
    do Gen.gen(x); do Print.print("abc"); generator(x + 1)

def sample(limit: Int32): List[Int32] \ Print =
    try {
        generator(0); Nil
    } with Gen {
        def gen(x, k) = if (x == limit) Nil else x :: k()
    }

def main(): Unit \ IO =
    try {
        println(sample(4))
    } with Print {
        def print(x, k) = { println(x); k() }
    }
```

I know its incomplete (i.e. it rejects programs with polymorphic handlers), but I am not sure if it is unsound. 
Possibly it could be sound still.

In any case, this is just a temporary measure while we wait for sets :)

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala [hunk 1]
```diff
@@ -512,6 +512,29 @@ object Unification {
...
+    */
+  def purifyEff(sym: Symbol.EffectSym, eff: Type): InferMonad[Type] = {
+    def visit(t: Type): Type = t match {
+      case Type.Var(_, _) => t
+      case Type.Cst(tc, _) => tc match {
+        case TypeConstructor.Effect(sym2) if sym == sym2 => Type.Pure
+        case _ => t
+      }
+      case Type.Apply(tpe1, tpe2, loc) => Type.Apply(visit(tpe1), visit(tpe2), loc)
+      case Type.Alias(cst, args, tpe, loc) => Type.Alias(cst, args.map(visit), visit(tpe), loc)
```
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala:528]: Why?

**Comment by magnus-madsen** [comment]
I have failing tests it seems :)

**Comment by magnus-madsen** [comment]
@mlutze Updated

File: main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala]: If I make an alias `MyThing = {Console, Network}` then we want `Console` to be deleted when it's handled.

**Review by mlutze — APPROVED** [review]

**Pull request merged**